ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogWorker.java

```diff
@@ -321,6 +329,9 @@ private void run() {
           }
           task.done();
         }
+        if (asyncFlushEnabled) {
+          flushToPersistentStore();
+        }
```

Comment:
The flush actually is sync since it is still run in the main thread.  It just runs outside the WriteLog task.

Reply:
Yes, it's true